samples/subsys/mgmt/mcumgr/smp_svr/src/bluetooth.c

```diff
@@ -58,9 +58,24 @@ BT_CONN_CB_DEFINE(conn_callbacks) = {
 	.disconnected = disconnected,
 };
 
+static void bt_ready(int err)
+{
+	if (err != 0) {
+		LOG_ERR("Bluetooth failed to initialise: %d", err);
+	} else {
+		k_work_submit(&advertise_work);
+	}
+}
+
 void start_smp_bluetooth_adverts(void)
 {
+	int rc;
+
 	k_work_init(&advertise_work, advertise);
+	rc = bt_enable(bt_ready);
 
 	k_work_submit(&advertise_work);
```

Comment:
Should we actually start advertising here? Shouldn't we wait for Bluetooth readiness (so that submit the work only from `bt_ready` callback?

Reply:
Good catch! Seems like a rebase error since the upstream code is correct:

https://github.com/zephyrproject-rtos/zephyr/commit/4516e7161c8e3ca93239e189741f8c60ce678fb3#

See the force push diff here: https://github.com/nrfconnect/sdk-zephyr/compare/0388cfd8f0612f8eb46308215c58756e58c3b5e8..04b9f936ebdfec8ed2c72c7fc05985463192b99a